frontend/demo/component/datepicker/react/date-picker-validation.tsx

```diff
@@ -15,14 +15,19 @@ function Example() {
     <DatePicker
       label="Appointment date"
       helperText="Must be within 60 days from today"
+      required
       min={formatISO(minDate.value, { representation: 'date' })}
       max={formatISO(maxDate.value, { representation: 'date' })}
       errorMessage={errorMessage.value}
-      onChange={({ target }) => {
-        const date = parse(target.value ?? '', 'yyyy-MM-dd', new Date());
-        if (isBefore(date, minDate.value)) {
+      onValidated={(event) => {
+        const field = event.target as DatePickerElement;
+        if (!field.value && (field.inputElement as HTMLInputElement).value) {
+          errorMessage.value = 'Invalid date format';
+        } else if (!field.value) {
+          errorMessage.value = 'Field is required';
```

Comment:
Maybe we could simplify this a bit to only check `!field.value` once:

```suggestion
        if (!field.value) {
          const notEmpty = (field.inputElement as HTMLInputElement).value;
          errorMessage.value = notEmpty ? 'Invalid date format' : 'Field is required';
```

Reply:
IMO, the current approach has a benefit that it keeps all error messages at the same indentation level, which makes them a little bit easier to scan.